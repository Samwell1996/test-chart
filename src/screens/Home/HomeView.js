import React, { memo } from 'react';
import { SafeAreaView, View, Text } from 'react-native';
import { getHeight } from './helper';
import s from './styles';

const HomeView = ({ item }) => {
  const value = item?.chart?.value;

  return (
    <View style={s.container}>
      <SafeAreaView>
        <Text style={s.text}>Diagram</Text>
        {value && (
          <>
            <View style={s.diagramContainer}>
              <View style={{ flex: getHeight(false, value) }} />
              <View style={[{ flex: getHeight(true, value) }, s.chart]} />
            </View>
            <Text style={s.text}>{value}</Text>
          </>
        )}
      </SafeAreaView>
    </View>
  );
};

export default memo(HomeView);
