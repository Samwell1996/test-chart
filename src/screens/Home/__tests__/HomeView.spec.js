import HomeView from '../HomeView';
import renderer from 'react-test-renderer';

test('render ui HomeScreen', () => {
  it('make snapshot for HomeScreen', () => {
    expect(HomeView).toMatchSnapshot();
  });
  it('renders HomeScreen correctly!', () => {
    renderer.create(<HomeView />);
  });
});
