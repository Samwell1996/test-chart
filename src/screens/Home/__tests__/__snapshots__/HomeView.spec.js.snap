// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ui Home screen 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "compare": null,
  "type": [Function],
}
`;
